etna/search/templatetags/search_tags.py

```diff
@@ -7,98 +7,12 @@
 from django.utils.crypto import get_random_string
 from django.utils.safestring import mark_safe
 
-import bleach
-
 from ...ciim.constants import SearchTabs
 
 register = template.Library()
 logger = logging.getLogger(__name__)
 
 
-@register.filter
-def record_detail(record: dict, key: str) -> str:
-    """Fetch an item from a record's details template.
-
-    Django templates don't allow access to keys or attributes prefixed with _
-
-    https://docs.djangoproject.com/en/4.0/ref/templates/language/#variables
-    """
-    try:
-        return record["_source"]["@template"]["details"][key]
-    except KeyError:
-        return ""
-
-
-def record_highlight(record: dict, key: str) -> str:
-    """
-    Fetch a key for a record, either from the highlight dict, or the details template and strip all HTML apart from <mark>.
-    """
-    try:
-        value = record["highlight"][f"@template.details.{key}"]
-    except KeyError:
-        try:
-            value = record["_source"]["@template"]["details"][key]
-        except KeyError:
-            return ""
-
-    if not value:
-        return ""
-    if isinstance(value, (list, tuple)):
-        value = value[0]
-
-    return mark_safe(bleach.clean(value, tags=["mark"], strip=True))
```

Comment:
This code is removed, but did not find it in [`def highlights`](https://github.com/nationalarchives/ds-wagtail/blob/19df21c949fde726b210e9da49f732a4442fea38/etna/records/models.py#L66) Search results will show without cleaning. Are we raising it to be cleaned in the API response with `mark` retained?

Reply:
I will investigate further into whether further work is needed here.

In regards to cleaning of values: I feel we really need to the API to decide on types of values it will serve, enforce that, and go from there. I don't see any good reason why any of the fields we are using would contain HTML (they just 'do'  - and we're having to 'hack around it')

Reply:
The one interpretive field that was being used in output has now been updated to strip any HTML in a similar way (see d6720a6)

Reply:
Please see example `/search/catalogue/?filter_keyword=historic&opening_start_date_0=&opening_start_date_1=&opening_start_date_2=&opening_end_date_0=&opening_end_date_1=&opening_end_date_2=&q=royal&per_page=20&sort_order=asc&display=list&group=tna`

In the above tag implementation data extracted here for highlight with <mark> is outside of `_source` i.e. `highlight@template.details.<key>` , therefore not getting <mark> in the output. For the same reason, I think the data extracted for `description` is not getting cleaned.

Reply:
Just investigating this issue now

Reply:
So, it seems there are a few different issues here:
1. Highlight data is being currently being looked for in the wrong place (`raw_data["_source"]["highlight"]`). This can be fixed by updating `Record.__init__()` to look in the correct place in `raw_data` (`raw_data["highlight"]`), and dropping the `Record.highlights` cached_property.
2 `Record.description` is being used as plain text in templates, so really, the raw values should be passed through `strip_html()` 
3. The data in 'highlights' (when we actually pull it in correctly) includes partial tag HTML in it's list of 'marked strings', which make them impossible to clean up. Realistically, the only solution here is for K-Int to remove any HTML from these values on indexing, so that highlights come back as plain text with `mark` tags added (as expected)

Reply:
The first issue above is resolved by 2581415 and 34a7ca6

Reply:
The second is resolved by ce8389f